tests/amr/data/field/refine/test_refine_field.py

```diff
@@ -0,0 +1,35 @@
+import numpy as np
+
+from pyphare.core import box as boxm
+from pyphare.core.gridlayout import GridLayout
+from pyphare.pharesee.hierarchy import FieldData
+
+
+# only supports ndim == 1, updates needed for > 1
+# assumes refinement_ratio is 2
+
+def refine(field, **kwargs):
+    assert isinstance(field, FieldData)
+    kwarg_keys = ["data"] # doing if data == None for input data=None, fails if it's an np array
+    assert any([key in kwarg_keys for key in list(kwargs.keys())])
+
+    refinement_ratio = 2
+    primal_directions = field.primal_directions()
+    data   = kwargs.get("data", field.dataset[:])
+    ghosts = field.ghosts_nbr
+
+    fine_box    = boxm.refine(field.box, refinement_ratio)
+    fine_layout = GridLayout(fine_box, field.origin, field.layout.dl/refinement_ratio, interp_order=field.layout.interp_order)
+
+    assert field.box.ndim == 1
+
+    fine_data = np.zeros(fine_box.shape + primal_directions + (field.ghosts_nbr[0] * 2))
+
+    if primal_directions[0]:
+        fine_data[5:-5:2] = data[5:-5] # coarse primal on top of fine
```

Comment:
I think you want to use the actual ghost nbr variable since you have it

Reply:
should the ":2" be refinement_factor or maybe "cadence = 2" or something?

Reply:
hmm yeah it's because of the refinement ratio, but if we were using a different ratio the formula would be different as well, no just that slicing...